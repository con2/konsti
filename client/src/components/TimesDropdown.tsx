import React, { FC, ReactElement, ChangeEvent } from 'react';
import { timeFormatter } from 'utils/timeFormatter';

export interface Props {
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
  selectedTime: string;
  times: readonly string[];
}

// TODO: Only enable next open signup
// Check current time and enable new timestamp
// Show "signup starts xx:xx" on others
// Toggle to show upcoming gameslots or all gameslots

export const TimesDropdown: FC<Props> = (props: Props): ReactElement => {
  const { times, onChange, selectedTime } = props;

  const sortedTimes = times.map((sortedTime) => {
    const formattedDate = timeFormatter.weekdayAndTime({
      time: sortedTime,
      capitalize: true,
    });
    return (
      <option value={sortedTime} key={sortedTime}>
        {formattedDate}
      </option>
    );
  });

  return (
    <div className='times-dropdown'>
      <select onChange={onChange} value={selectedTime}>
        {sortedTimes}
      </select>
    </div>
  );
};
