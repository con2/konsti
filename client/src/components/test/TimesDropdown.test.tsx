import React from 'react';
import { shallow } from 'enzyme';
import { TimesDropdown, Props } from '../TimesDropdown';

describe('TimesDropdown', () => {
  it('should render correctly', () => {
    const props: Props = { times: [], onChange: () => {}, selectedTime: '' };
    const component = shallow(<TimesDropdown {...props} />);
    expect(component).toMatchSnapshot();
  });
});
