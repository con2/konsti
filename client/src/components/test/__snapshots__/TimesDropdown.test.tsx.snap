// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimesDropdown should render correctly 1`] = `
<div
  className="times-dropdown"
>
  <select
    onChange={[Function]}
    value=""
  />
</div>
`;
